examples/scene-example.ts

```diff
@@ -0,0 +1,378 @@
+
+import { mat4, vec3 } from 'gl-matrix';
+
+import {
+    Box,
+    Camera,
+    Canvas,
+    Context,
+    DefaultFramebuffer,
+    ForwardSceneRenderPass,
+    Framebuffer,
+    GeometryComponent,
+    Invalidate,
+    Material,
+    MouseEventProvider,
+    Navigation,
+    Program,
+    Renderer,
+    SceneNode,
+    Shader,
+    Sphere,
+    Texture2D,
+    Wizard,
+} from 'webgl-operate';
+
+import { Example } from './example';
+
+// tslint:disable:max-classes-per-file
+
+/**
+ * @todo comment
+ */
+export class SceneRenderer extends Renderer {
+
+    protected _navigation: Navigation;
+
+    protected _forwardPass: ForwardSceneRenderPass;
+
+    protected _camera: Camera;
+
+    protected _scene: SceneNode;
+
+    protected _texture: Texture2D;
+    protected _framebuffer: Framebuffer;
+
+    /**
+     * Initializes and sets up rendering passes, navigation, loads a font face and links shaders with program.
+     * @param context - valid context to create the object for.
+     * @param identifier - meaningful name for identification of this instance.
+     * @param mouseEventProvider - required for mouse interaction
+     * @returns - whether initialization was successful
+     */
+    protected onInitialize(context: Context, callback: Invalidate,
+        mouseEventProvider: MouseEventProvider,
+        /* keyEventProvider: KeyEventProvider, */
+        /* touchEventProvider: TouchEventProvider */): boolean {
+
+        this._framebuffer = new DefaultFramebuffer(this._context, 'DefaultFBO');
+        this._framebuffer.initialize();
+
+        /* Create and configure camera. */
+
+        this._camera = new Camera();
+        this._camera.center = vec3.fromValues(0.0, 0.0, 0.0);
+        this._camera.up = vec3.fromValues(0.0, 1.0, 0.0);
+        this._camera.eye = vec3.fromValues(0.0, 0.0, 3.0);
+        this._camera.near = 0.1;
+        this._camera.far = 8.0;
+
+        /* Create and configure navigation */
+
+        this._navigation = new Navigation(callback, mouseEventProvider);
+        this._navigation.camera = this._camera;
+
+        /* Create and configure forward pass. */
+
+        this._forwardPass = new ForwardSceneRenderPass(context);
+        this._forwardPass.initialize();
+
+        this._forwardPass.camera = this._camera;
+        this._forwardPass.target = this._framebuffer;
+
+        /* Create scene. */
+
+        this.generateScene();
+        this._forwardPass.scene = this._scene;
+
+        return true;
+    }
+
+    /**
+     * Uninitializes Buffers, Textures, and Program.
+     */
+    protected onUninitialize(): void {
+        super.uninitialize();
+
+        // TODO: make sure that all meshes and programs inside of the scene get cleaned
```

Comment:
Which components needs to be uninitialized?
Are all member variables that are initialized during `onInitialize` required to be undefined?